prov/efa/test/efa_unit_test_mocks.c

```diff
@@ -0,0 +1,72 @@
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include "efa.h"
+
+void __real_rxr_pkt_handle_send_completion(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry);
+
+void __wrap_rxr_pkt_handle_send_completion(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry)
+{
+	check_expected(ep);
+	check_expected(pkt_entry);
+
+	if ((int)mock() == 42) {
+		__real_rxr_pkt_handle_send_completion(ep, pkt_entry);
+	}
+}
+
+void __real_rxr_pkt_handle_recv_completion(struct rxr_ep *ep,
+										   struct rxr_pkt_entry *pkt_entry,
+										   enum rxr_lower_ep_type lower_ep_type);
+
+void __wrap_rxr_pkt_handle_recv_completion(struct rxr_ep *ep,
+										   struct rxr_pkt_entry *pkt_entry,
+										   enum rxr_lower_ep_type lower_ep_type)
+{
+	check_expected(ep);
+	check_expected(pkt_entry);
+	check_expected(lower_ep_type);
+
+	if ((int)mock() == 42) {
```

Comment:
Because of [memes](https://www.buzzfeed.com/australianredcrossbloodservice/definitive-proof-number-42-actually-is-the-answer-to-life). I have another idea in mind w.r.t. mocks in general. I put 42 as a placeholder for now. Will refactor the mocks later.

It shouldn't matter unless someone want to explicitly return `FI_ENOMSG`.

Reply:
Maybe use a macro like EFA_UNIT_TEST_CALL_REAL, and change the value to 4242.

> someone want to explicitly return FI_ENOMSG.

This can happen in some test case.

Reply:
Ok I will hard code it to 4242 for now.